fdbserver/RestoreMaster.actor.cpp

```diff
@@ -862,4 +904,67 @@ ACTOR static Future<Void> signalRestoreCompleted(Reference<RestoreMasterData> se
 	TraceEvent("FastRestore").detail("RestoreMaster", "AllRestoreCompleted");
 
 	return Void();
+}
+
+// Update the most recent time when master receives hearbeat from each loader and applier
+ACTOR static Future<Void> updateHeartbeatTime(Reference<RestoreMasterData> self) {
+	int numRoles = self->loadersInterf.size() + self->appliersInterf.size();
+	state std::map<UID, RestoreLoaderInterface>::iterator loader = self->loadersInterf.begin();
+	state std::map<UID, RestoreApplierInterface>::iterator applier = self->appliersInterf.begin();
+	state std::vector<Future<RestoreCommonReply>> fReplies(numRoles, Never()); // TODO: Reserve memory for this vector
+	state std::vector<UID> nodes;
+	state int index = 0;
+
+	// Initialize nodes only once
+	loader = self->loadersInterf.begin();
+	applier = self->appliersInterf.begin();
+	while (loader != self->loadersInterf.end()) {
+		nodes.push_back(loader->first);
+		loader++;
+	}
+	while (applier != self->appliersInterf.end()) {
+		nodes.push_back(applier->first);
+		applier++;
+	}
+
+	loop {
+		wait(delay(SERVER_KNOBS->FASTRESTORE_HEARTBEAT_DELAY));
```

Comment:
line 948 also delays amount. So it's possible to have double delays.

Reply:
yeah, I realized this while I was writing. I felt it was not so important. But I can do better to avoid this and I will. :)